### PR TITLE
Transactional auth

### PR DESCRIPTION
**Pull request by marek-vysoky**
Adding implementation, exchenges token.
Adding example project

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @marek-vysoky has signed the CLA. Thanks for the pull request!

File: Box.V2.sln [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ﻿
 Microsoft Visual Studio Solution File, Format Version 12.00
-# Visual Studio 14
-VisualStudioVersion = 14.0.23107.0
+# Visual Studio 2013
```
Comment (coolboy) [inline Box.V2.sln]: What changed it back to an older version?

File: Box.V2.Samples.TransactionalAuth/Program.cs [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            t.Wait();
+
+            Console.WriteLine();
+            Console.Write("Press return to exit...");
+            Console.ReadLine();
+
+        }
+        private static async Task MainAsync(string fileId, bool usePrimaryToken)
+        {
+
```
Comment (coolboy) [inline Box.V2.Samples.TransactionalAuth/Program.cs]: useless empty line, should be removed

File: Box.V2.Samples.TransactionalAuth/Program.cs [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+            string resource = string.Format("https://api.box.com/2.0/files/{0}", fileId);
+            var boxTransactional = new BoxTransactionalAuth();
+            string token = usePrimaryToken ? PRIMARY_TOKEN : SECONDARY_TOKEN;
+            var client = boxTransactional.GetClient(token, resource);
+            var fileInfo = await client.FilesManager.GetInformationAsync(fileId);
+            if (fileInfo.Id == fileId)
+            {
+                Console.WriteLine(string.Format("File name is {0} ", fileInfo.Name));
+            }
+
```
Comment (coolboy) [inline Box.V2.Samples.TransactionalAuth/Program.cs]: empty lines should be removed

File: Box.V2/Models/OAuthSession.cs [hunk 4]
```diff
@@ -71,9 +73,15 @@ public OAuthSession(string access_token, string refresh_token, int expires_in, s
...
+   
+        /// <summary>
+        /// Gets the restricted to.
+        /// </summary>
+        /// <value>
+        /// The restricted to.
+        /// </value>
         [JsonProperty(PropertyName = FieldRestrictedTo)]
-        public string[] RestrictedTo { get; private set; }
+        public List<BoxRestrictedTo> RestrictedTo { get; private set; }
```
Comment (coolboy) [inline Box.V2/Models/OAuthSession.cs]: Is this a public interface change, won't it break our customers' code?

**Comment by marek-vysoky** [comment]
I've removed useless empty lines, I've returned back RestrictedTo type to string[]

**Comment by marek-vysoky** [comment]
I've returned version of Visual studio in Box.V2.sln

**Comment by cburnette** [comment]
@coolboy I'll definitely need to fully vet this one as Transactional Auth is new and I don't yet understand all the details.

**Pull request merged**